### PR TITLE
OBP-289 - Migrate create-tag-index.rb to bulk indexer task

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: docs/Virtuoso.md [hunk 1]
```diff
@@ -202,7 +202,3 @@ SQL> quit;
 > exit
 > logout
 ```
-
```
Comment (paulpilone) [inline docs/Virtuoso.md:205]: Can we just get rid of this file now?

File: ingest/bin/ingest-rectifier.rb [hunk 2]
```diff
@@ -9,7 +9,7 @@
 
 options = {}
 OptionParser.new do |opts|
-  opts.banner = "Usage: create-tag-index.rb [options]"
```
Comment (paulpilone) [inline ingest/bin/ingest-rectifier.rb:12]: Don't worry about it now but this file can be removed after my big feature is finished - I've implemented this feature as a Lambda function.

File: lib/open-search-client.js [hunk 3]
```diff
@@ -195,7 +195,7 @@ const createTermsIndex = (prefixUrl, index) => createIndex(prefixUrl, index, {
       source_terminology: {
         type: 'keyword',
       },
-      graphUri: {
+      ontologyGraph: {
```
Comment (paulpilone) [inline lib/open-search-client.js]: Looks like this is probably the merge conflict. Any particular reason for this change?

File: docs/Virtuoso.md (hunk 1 above)
Comment (ghost) [reply docs/Virtuoso.md:205]: Done in bdc4199

File: neptune-bulk-loader/task.ts [hunk 4]
```diff
@@ -54,10 +55,19 @@ export const neptuneBulkLoader = async (): Promise<MainResult> => {
...
-      graphUri: metadata.namedGraphUri,
-    },
+    match: { ontologyGraph },
+  });
+
+  await indexTerms({
+    elasticsearchUrl: esUrl,
+    ontologyNameSpace: metadata.ontologyNameSpace,
+    ontologyGraph,
+    terminologyTitle: 'my-terminology-title',
```
Comment (paulpilone) [inline neptune-bulk-loader/task.ts]: Should this come from the metadata?

Comment (paulpilone) [reply neptune-bulk-loader/task.ts]: It's not part of the metadata schema right now but how else would we get it? Either it has to come from the actual ontology (we'd need a query for that) or we just ask them to provide it as part of the metadata (like the namespace).

File: neptune-bulk-loader/index-terms.ts [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+
+  if (response.statusCode !== 200) {
+    throw new Error(`Neptune request failed with status ${response.statusCode}: ${response.body}`);
+  }
+};
+
+interface CreateTermIndexParams {
+  elasticsearchUrl: string
+  ontologyNameSpace: string
+  ontologyGraph: string
```
Comment (paulpilone) [inline neptune-bulk-loader/index-terms.ts]: This is a small thing but in this file it's called `ontologyGraph` but it comes from `ontologyGraphUrl` - which I think in Neptune is actually `namedGraphUri`). It doesn't really matter what it's called just would like it to be consistent - - unless there is a reason you've made them different.

Comment (ghost) [reply neptune-bulk-loader/index-terms.ts]: Fixed in 33fae9e

File: lib/open-search-client.js (hunk 3 above)
Comment (marchuffnagle) [reply lib/open-search-client.js]: Updated in 33fae9e

File: neptune-bulk-loader/task.ts (hunk 4 above)
Comment (ghost) [reply neptune-bulk-loader/task.ts]: This was fixed in e13e1f5

**Review by paulpilone — APPROVED** [review]

**Pull request merged**